plugins/cc/ccalgo.ml

```diff
@@ -207,7 +207,15 @@ struct
     constr = lazy (constr_of_term t);
     hash = hash_term t }
 
-  let mkSymb s = make (Symb s)
+  let rec drop_univ c =
+    match kind c with
+    | Const (c,_u) -> mkConstU (c,Univ.Instance.empty)
+    | Ind (c,_u) -> mkIndU (c,Univ.Instance.empty)
+    | Construct (c,_u) -> mkConstructU (c,Univ.Instance.empty)
+    | Sort (Type _u) -> mkSort (type1)
+    | _ -> Constr.map drop_univ c
+
+  let mkSymb s = make (Symb (s,drop_univ s))
```

Comment:
Maybe we could store just the hash instead of keeping an invalid constr around, ie
```suggestion
  let mkSymb s = make (Symb (s,Constr.hash (drop_univ s)))
```
or even define a hash_nounivs

Reply:
Indeed, much cleaner.

> or even define a hash_nounivs

Also possible. No strong opinion in either way.